omegaconf/grammar/OmegaConfGrammarLexer.g4

```diff
@@ -16,13 +16,23 @@ fragment ESC_BACKSLASH: '\\\\';  // escaped backslash
 
 TOP_INTER_OPEN: INTER_OPEN -> type(INTER_OPEN), pushMode(INTERPOLATION_MODE);
 
-ESC_INTER: '\\${';
-TOP_ESC: ESC_BACKSLASH+ -> type(ESC);
+// Regular string: anything that does not contain any $ and does not end with \
+// (this ensures this rule will not consume characters required to recognize other tokens).
+ANY_STR: ~[$]* ~[\\$];
+
+// Escaped interpolation: '\${', optionally preceded by an even number of \
+ESC_INTER: ESC_BACKSLASH* '\\${';
+
+// Escaped backslashes: even number of \ followed by an interpolation.
+// The interpolation must *not* be matched by this rule (this is why we use a predicate lookahead).
+TOP_ESC: ESC_BACKSLASH+ { self._input.LA(1) == ord("$") and self._input.LA(2) == ord("{") }? -> type(ESC);
```

Comment:
This is the first time we are adding custom code to ANTLR.
This means the lexer code is no longer portable and cannot be used to generate non-python code.
Is this strictly necessary or can we offload the logic to the visitor?

Reply:
See discussion in the comment below: https://github.com/omry/omegaconf/pull/708#issuecomment-833833406

Reply:
Following our chat, I moved that logic to the visitor in c5d3a07